source/common/tcp_proxy/upstream.cc

```diff
@@ -278,13 +278,18 @@ void Http2Upstream::setRequestEncoder(Http::RequestEncoder& request_encoder, boo
   auto headers = Http::createHeaderMap<Http::RequestHeaderMapImpl>({
       {Http::Headers::get().Method, config_.usePost() ? "POST" : "CONNECT"},
       {Http::Headers::get().Host, config_.hostname()},
-      {Http::Headers::get().Path, "/"},
-      {Http::Headers::get().Scheme, scheme},
   });
 
-  if (!config_.usePost()) {
-    headers->addReference(Http::Headers::get().Protocol,
-                          Http::Headers::get().ProtocolValues.Bytestream);
+  if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.use_rfc_connect")) {
```

Comment:
super nit but it might be nice to have all these checks have the feature enabled path first, just to make it easier to follow along the two branches when reading